api/migrations/env.py

```diff
@@ -17,24 +15,45 @@
 fileConfig(config.config_file_name)
 logger = logging.getLogger("alembic.env")
 
+
+def get_engine():
+    try:
+        # this works with Flask-SQLAlchemy<3 and Alchemical
+        return current_app.extensions["migrate"].db.get_engine()
+    except TypeError:
+        # this works with Flask-SQLAlchemy>=3
+        return current_app.extensions["migrate"].db.engine
+
+
+def get_engine_url():
+    # Set the database URL directly from environment variables.
+    return f"postgresql://{os.environ.get('DB_USERNAME', 'admin')}:{os.environ.get('DB_PASSWORD', 'development')}@{os.environ.get('DB_HOST', 'pg')}/{os.environ.get('DATABASE_NAME', 'api_development')}"
```

Comment:
This ensures that flask-migrate / Alembic read from the same database environment variables that docker-compose sets.